web/src/copy_and_paste.ts

```diff
@@ -472,6 +489,122 @@ export function paste_handler_converter(paste_html: string): string {
+                        consecutive_empty_display_count = 0;
+                    } else {
+                        // Handling cases where math block is selected directly without any preceding text.
+                        // The initial katex display doesn't have the annotation when selection is done in this manner.
+                        if (
+                            child.classList.contains("katex-display") &&
+                            child.querySelector("math")?.textContent !== ""
+                        ) {
+                            math_block_markdown += content;
+                            continue;
+                        }
+                        if (consecutive_empty_display_count === 0) {
+                            math_block_markdown += "\n\n\n";
+                        } else {
+                            math_block_markdown += "\n\n";
+                        }
+                        consecutive_empty_display_count += 1;
+                    }
+                }
+            }
+            math_block_markdown = math_block_markdown.slice(0, -1); // Don't add extra newline add the end
+            return (math_block_markdown += "```");
+        },
+    });
+
+    turndownService.addRule("katex-inline-math", {
+        filter(node) {
+            // Allow the intermediate text blocks, so that they don't get appended to the end.
+            if (node.classList.contains("intermediate-katex-text-node")) {
+                return true;
+            }
+            if (node.classList.contains("katex") && !node.classList.contains("katex-display")) {
+                // Should explicitly be an inline expression
+                const parent = node.parentElement;
+                if (parent?.classList.contains("katex-display")) {
+                    return false;
+                }
+                if (
+                    parent?.nodeName !== "P" &&
+                    node.querySelector(".katex-mathml annotation")?.textContent === null
+                ) {
+                    return false;
+                }
+                return true;
+            }
+
+            return false;
+        },
+        replacement(content, node: Node) {
+            let parsed_inline_expression = "";
+            if (node !== null) {
+                const parent = node.parentElement!;
+                if (processed_inline_math_block_parents.has(parent)) {
+                    return "";
+                }
+                processed_inline_math_block_parents.add(parent);
+                for (const child of parent.children) {
+                    if (child.nodeName === "BR") {
+                        parsed_inline_expression += "\n";
+                        continue;
+                    }
+                    const annotation_element = child.querySelector(
+                        `.katex-mathml annotation[encoding="application/x-tex"]`,
+                    );
+                    if (!annotation_element?.textContent) {
```

Comment:
I feel like flipping this conditional would be more readable -- you check if this value exists and then immediately use it inside, rather than having the code involving `annotation_element` be so spread apart.